app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,481 @@
+        username: currentUser.username,
+    });
+    const [canSave, setCanSave] = useState(false);
+    const [error, setError] = useState<ErrorText | undefined>();
+    const [updating, setUpdating] = useState(false);
+
+    const scrollViewRef = useRef<KeyboardAwareScrollView>();
+
+    const buttonText = intl.formatMessage({id: 'mobile.account.settings.save', defaultMessage: 'Save'});
+    const rightButton = useMemo(() => {
+        return isTablet ? null : {
+            id: 'update-profile',
+            enabled: false,
+            showAsAction: 'always' as const,
+            testID: 'edit_profile.save.button',
+            color: theme.sidebarHeaderTextColor,
+            text: buttonText,
+        };
+    }, [isTablet, theme.sidebarHeaderTextColor]);
+
+    const closeButtonId = 'close-edit-profile';
+    const service = currentUser.authService;
+
+    const leftButton = useMemo(() => {
+        return isTablet ? null : {
+            id: closeButtonId,
+            icon: CompassIcon.getImageSourceSync('close', 24, theme.centerChannelColor),
+            testID: closeButtonId,
+        };
+    }, [isTablet, theme.sidebarHeaderTextColor]);
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case 'update-profile':
+                        submitUser();
+                        break;
+                    case closeButtonId:
+                        close();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [userInfo]);
+
+    useEffect(() => {
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+        return () => {
+            backHandler.remove();
+        };
+    }, []);
+
+    useEffect(() => {
+        if (!isTablet) {
+            setButtons(componentId, {
+                rightButtons: [rightButton!],
+                leftButtons: [leftButton!],
+            });
+        }
+    }, []);
+
+    const close = useCallback(() => {
+        if (isModal) {
+            dismissModal({componentId});
+        } else if (isTablet) {
+            DeviceEventEmitter.emit(Events.ACCOUNT_SELECT_TABLET_VIEW, '');
+        } else {
+            popTopScreen(componentId);
+        }
+
+        return true;
+    }, []);
+
+    const enableSaveButton = useCallback((value: boolean) => {
+        if (!isTablet) {
+            const buttons = {
+                rightButtons: [{
+                    ...rightButton!,
+                    enabled: value,
+                }],
+            };
+            setButtons(componentId, buttons);
+        }
+        setCanSave(value);
+    }, [componentId, rightButton]);
+
+    const submitUser = useCallback(preventDoubleTap(async () => {
+        enableSaveButton(false);
```

Comment:
If the right button changes (e.g. theme changes) and then we submit the user, the enableSaveButton used in this callback will be the old one, setting the right button as the old theme.

To "fix" this, add enableSaveButton to the dependencies of this callback.

Reply:
`enableSaveButton` has a dependency array on `rightButton`.... `rightButton` has dependency on theme.  I think we are covered, right?

Reply:
`enableSaveButton` and `rightButton` update just fine. The problem is how often `submitUser` is recalculated.

For the effect that depends on `userInfo` there is no problem, since `submitUser` also depends on `userInfo` and will be recreated with the latest `enableSaveButton` and `rightButton`.

But you have two other places where `submitUser` is being used: one in `onFocusNextField` (haven't yet looked down that rabbit hole) depending on `canSave`, and the TableTitle component during render.

So... the reproduction steps would be something like...
- Component renders for the first time while theme is blue.
- Right button is blue, and enableSaveButton is prepared to enable a blue button. SubmitUser uses the enableSaveButton that put a white button.
- Theme changes to red (by websocket event, for example)
- Right button is updated to be red. enableSaveButton is prepared to enable a red button. SubmitUser is not updated, so still uses the enableSaveButton with blue theme.
- I submit the user.
- I show for a brief time a red disabled right button.
- If there is an error, we would still be using also the old `resetScreen` function (submitUser has not been updated in all this time). So we will show a red enabled right button.

It is, without any doubt, a extreme corner case (theme changing in the middle of something) but may happen.

Reply:
Done